scripts/start_server_and_test

```diff
@@ -14,12 +14,14 @@ START_SERVER_SCRIPT="$1"
 PORT="$2"
 TEST_SCRIPT="$3"
 
-await_port --await-free "$PORT"
+await_port --await-free --timeout 120 "$PORT"
 
 eval "$START_SERVER_SCRIPT &"
 SERVER_PID=$!
 
 function finish {
+  SERVER_PID=`lsof -i -P -n | grep LISTEN | grep ${PORT} | awk '{ print $2 }'`
+  echo Killing server PID=${SERVER_PID}
```

Comment:
the pid of the `eval` was not (always?) the web server pid, so killing that pid didn't work.

Reply:
ah, I had just added a hackier solution to this by moving to `kill -9`. This is much prettier. (the more pipes the better, right? :trollface: ).

It only just dawned on me a couple hours ago that this was the problem; I didn't realize the server process could be orphaned from python... 😞 ... and eventually my machine would clean up the process, sometimes after 30s, sometimes after 5 min. 🤷‍♂️

Reply:
but this was annoying 😅

Reply:
I didn't investigate too deeply, but it seemed like killing the parent process (the `make` process, in this case) kills the child process (the web server) when running on MacOS, but not Ubuntu. Or maybe it never worked on either...

Reply:
😢

Reply:
it definitely failed on MacOS half the time and was successful half the time...